community-tutorials/ubuntu-docker-install/01-en.md

```diff
@@ -0,0 +1,280 @@
+
+-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 192.168.0.0/16
+-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 10.0.0.0/8
+-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 172.16.0.0/12
+-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 192.168.0.0/16
+-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 10.0.0.0/8
+-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 172.16.0.0/12
+
+-A DOCKER-USER -j RETURN
+
+-A ufw-docker-logging-deny -m limit --limit 3/min --limit-burst 10 -j LOG --log-prefix "[UFW DOCKER BLOCK] "
+-A ufw-docker-logging-deny -j DROP
+
+COMMIT
+# END UFW AND DOCKER
+```
+
+After saving the modifications restart the server to have the changes take effect.
+
+```bash
+sudo reboot
+```
+
+
+## Step 3.3 - (Using ufw to open ports)
+When you want to publish a port from your containers use the following command:
+
+> This command publishes port 80 from any container's local port. Replace 80 with your custom ports for tcp communication. To use the UDP Protocol replace ``proto tcp`` with ``proto udp``.
+
+```bash
+ufw route allow proto tcp from any to any port 80
+```
+
+When you have multiple containers using the same port you can specify the container by its private (internal) ip:
+
+```bash
+ufw route allow proto tcp from any to 172.17.0.2 port 80
+```
+
+> **Please Note:** the port used at the end of this commands is the **internal** port of the container! You dont need to expose them to the host.
+
+## Step 3.4 - (Using the ufw-docker tool for easier setup)
+Instead of doing the above manual steps to install / configure ufw you could use an handy script provided by the authors of [ufw-docker Repository](https://github.com/chaifeng/ufw-docker).
+
+To install the tool run the following command:
+
+```bash
+sudo wget -O /usr/local/bin/ufw-docker \
+  https://github.com/chaifeng/ufw-docker/raw/master/ufw-docker
+sudo chmod +x /usr/local/bin/ufw-docker
+```
+
+To automatically apply the ufw ``after.rules`` change run:
+```bash
+sudo ufw-docker install
+```
+
+To publish a containers port use the command:
+```bash
+sudo ufw-docker allow CONTAINER_NAME PORT
+```
+
+Further information and help can be shown by using:
+```bash
+sudo ufw-docker help
+```
+
+# Step 4 - (Add your user to docker group to use Docker without sudo) (Optional)
+If you wish to use docker with your regular user without using ``sudo`` all the time you can add your user to the docker group.
+**Please bear in mind that users could exploit docker to gain root access to your host machine!**
+
+To add your user to docker group run:
+```bash
+# Create the docker group in case it not yet exist
+sudo groupadd docker
+# Add your user to the docker group
+sudo usermod -aG docker $USER
+```
+
+You need to log out and log back in to have the changes take effect.
```

Comment:
Could you please add how to check if it also worked?
I have tested with `docker run --rm hello-world`